src/dbnode/integration/wide_query_test.go

```diff
@@ -98,124 +95,62 @@ func buildExpectedWideEntryByShard(
 		}
 
 		found := false
-		for idx, sharded := range shardeEntries {
+		for idx, sharded := range shardedEntries {
 			if shard != sharded.shard {
 				continue
 			}
 
 			found = true
-			shardeEntries[idx].entries = append(sharded.entries, entry)
+			shardedEntries[idx].entries = append(sharded.entries, entry)
 			break
 		}
 
 		if found {
 			continue
 		}
 
-		shardeEntries = append(shardeEntries, shardedIndexEntry{
+		shardedEntries = append(shardedEntries, shardedWideEntry{
 			shard:   shard,
 			entries: []schema.WideEntry{entry},
 		})
 	}
 
-	sort.Slice(shardeEntries, func(i, j int) bool {
-		return shardeEntries[i].shard < shardeEntries[j].shard
+	sort.Slice(shardedEntries, func(i, j int) bool {
+		return shardedEntries[i].shard < shardedEntries[j].shard
 	})
 
 	var entries []schema.WideEntry
-	for _, sharded := range shardeEntries {
+	for _, sharded := range shardedEntries {
 		entries = append(entries, sharded.entries...)
 	}
 
+	// NB: IDs should only be included for documents that conclude a batch.
+	l := len(entries)
+	if l == 0 {
+		return entries
+	}
+
```

Comment:
Either something got broken here, or this `if` can be removed (together with the comment?).